### PR TITLE
Move SSA_stept definition to its own header file

### PR DESCRIPTION
**Pull request by romainbrenguier**
This makes it easier to find it, and makes reading the
`symex_target_equationt` class and this one easier.

No functional changes.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [na] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [na] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [na] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by peterschrammel — APPROVED** [review]

File: src/goto-symex/ssa_step.h [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+/// Function calls store `called_function` name as well as a vector of
+/// arguments `ssa_function_arguments`. The `converted` version of a
+/// variable will contain its version for the SAT/SMT conversion.
+class SSA_stept
+{
+public:
+  symex_targett::sourcet source;
+  goto_trace_stept::typet type;
+
+  // NOLINTNEXTLINE(whitespace/line_length)
```
Comment (tautschnig) [inline src/goto-symex/ssa_step.h]: This code has been reformatted, and thus these linter exclusions are no longer appropriate. Please remove all of them.

Comment (romainbrenguier) [reply src/goto-symex/ssa_step.h]: done. I could also reformat `symex_target_equation.h` before moving the code if you think that would make it easier to review

Comment (tautschnig) [reply src/goto-symex/ssa_step.h]: Ah, no, don't bother, I wasn't worried about that. I was just surprised seeing those linter exclusions where they seemingly didn't make sense, and then figured out that they were only necessary under the previous formatting.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 2eea4928e04aac193cc58650a19348789130aeb4).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/104375075

**Review by LAJW — APPROVED** [review]

**Pull request merged**